heat/core/logical.py

```diff
@@ -519,6 +519,9 @@ def sanitize_input_type(
     # if one of the tensors is distributed, unsplit/gather it
     if x.split is not None and y.split is None:
         t1 = manipulations.resplit(x, axis=None)
+        local_start = x.comm.chunk(t1.gshape, x.split)[0]
+        local_end = local_start + x.lshape[x.split]
+        y = y[local_start:local_end]
```

Comment:
When working with DNDarrays, the arguments to a function should be the same on all processes. For local operations on each process, use the methods _larray()_ or _lloc()_ to work with the underlying PyTorch tensors.